### DISCUSSION TITLE
How multi-vector search changes the reranking strategy

### DISCUSSION THREAD

**Discussion by YYGe01** [milvus-io/milvus#32741] [Q&A and General discussion]
I want to take the maximum value of multiple vector similarity matches as the final score.
Currently I only see WeightedRanker and RRFRanker. 
I hope I can customize my Ranker method.

**Comment by yhmo** [comment]
I suppose this is a new feature.
What kind of ranker do you want?

**Comment by YYGe01** [comment]
> I suppose this is a new feature.我想这是一个新功能。 What kind of ranker do you want? 您想要什么样的排名？

like this.
https://discuss.elastic.co/t/how-to-modify-default-scoring-calculation-when-using-multiple-knn-mixed-searches/356513
For example, I store a piece of text in three text vectors, and I want to take the maximum value of the three vectors as the final score.

**Reply by yhmo** [reply]
The link of elasic shows a weight rank of vector search and text match.
Milvus only supports rank of vector search.

"take the maximum value of the three vectors" ---------- what is "maximum value of the thress vectors"?

**Reply by YYGe01** [reply]
Official documentation includes this example.
https://milvus.io/docs/multi-vector-search.md

```
request_1 = AnnSearchRequest(**search_param_1) 
request_2 = AnnSearchRequest(**search_param_2) 
reqs = [request_1, request_2] 
rerank = WeightedRanker(0.8, 0.2)  
```
The final score should be: 0.8 * request_1_score + 0.2 * request_2_score

I would like the rerank to be:

```
request_1 = AnnSearchRequest(**search_param_1) 
request_2 = AnnSearchRequest(**search_param_2) 
reqs = [request_1, request_2] 
rerank = MAX(reqs)
```
The final score is: max(request_1_score, request_2_score)

**Reply by yhmo** [reply]
Ok, I see. I think this is a new Ranker type that can be implemented. 
Wait @xiaofan-luan comment.

**Reply by yiwen92** [reply]
@YYGe01  will you ensure that request 1 and 2 are using the same metric type?

**Comment by Olimoyo** [comment]
Is this still being considered as a feature? Alternatively, can we write our own custom ranker?

**Reply by xiaofan-luan** [reply]
> Is this still being considered as a feature? Alternatively, can we write our own custom ranker?

Please be patience and wait for milvus 2.6 comes out.
We have plan to support udf based reranker, and also rerank by scalar field, lke time.
